extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1813,3 +1783,104 @@ export class Bind {
     }
   }
 }
+
+class BindWalker {
+  /**
+   * @param {!Node} root
+   */
+  constructor(root) {
+    const doc = devAssert(
+      root.nodeType == Node.DOCUMENT_NODE ? root : root.ownerDocument,
+      'ownerDocument is null.'
+    );
+
+    const useQuerySelector = doc.documentElement.hasAttribute(
+      'i-amphtml-binding'
+    );
+    /** @private @const {boolean} */
+    this.useQuerySelector_ = useQuerySelector;
+
+    /** @type {!Node} */
+    this.currentNode = root;
+
+    /** @private {number} */
+    this.index_ = 0;
+
+    /** @private @const {!Array<!Element>} */
+    this.nodeList_ = useQuerySelector
+      ? toArray(root.querySelectorAll('[i-amphtml-binding]'))
+      : [];
+
+    // Confusingly, the old TreeWalker hit the root node. We need to match that behavior.
```

Comment:
Did it have any effect? I.e. is it harmful if we start skipping?

Reply:
No tests fail, but I don't know if we just never tested this case.